catalog/app/containers/Bucket/PackageUpdateDialog.js

```diff
@@ -0,0 +1,991 @@
+  }
+
+  const del = (path) => {
+    pipeValue(R.evolve({ deleted: R.assoc(path, true) }))
+  }
+
+  const restore = (path) => {
+    pipeValue(R.evolve({ deleted: R.dissoc(path) }))
+  }
+
+  const handleAction = (handler, ...args) => (e) => {
+    e.stopPropagation()
+    if (disabled) return
+    handler(...args)
+  }
+
+  const revertFiles = React.useCallback(() => {
+    if (disabled) return
+    onInputChange(meta.initial)
+    setUploads({})
+  }, [disabled, onInputChange, setUploads, meta.initial])
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({ onDrop })
+
+  const totalProgress = React.useMemo(() => getTotalProgress(uploads), [uploads])
+
+  const computedEntries = useMemoEq(value, ({ added, deleted, existing }) => {
+    const p1 = Object.entries(existing).map(([path, { size }]) => {
+      if (path in deleted) {
+        return { type: 'deleted', path, size }
+      }
+      if (path in added) {
+        const a = added[path]
+        return { type: 'modified', path, size: a.size, delta: a.size - size }
+      }
+      return { type: 'unchanged', path, size }
+    })
+    const p2 = Object.entries(added).reduce((acc, [path, { size }]) => {
+      if (path in existing) return acc
+      return acc.concat({ type: 'added', path, size })
+    }, [])
+    return R.sortBy((x) => [TYPE_ORDER.indexOf(x.type), x.path], p1.concat(p2))
+  })
+
+  const stats = useMemoEq(value, ({ added, deleted, existing }) => ({
+    added: Object.values(added).reduce(
+      (acc, f) => R.evolve({ count: R.inc, size: R.add(f.size) }, acc),
+      { count: 0, size: 0 },
+    ),
+    deleted: Object.keys(deleted).reduce(
+      (acc, path) => R.evolve({ count: R.inc, size: R.add(existing[path].size) }, acc),
+      { count: 0, size: 0 },
+    ),
+  }))
+
+  return (
+    <div className={classes.root}>
+      <div className={classes.header}>
+        <div
+          className={cx(
+            classes.headerFiles,
+            disabled // eslint-disable-line no-nested-ternary
+              ? classes.headerFilesDisabled
+              : error // eslint-disable-line no-nested-ternary
+              ? classes.headerFilesError
+              : warn
+              ? classes.headerFilesWarn
+              : undefined,
+          )}
+        >
+          Files
+          {(!!stats.added.count || !!stats.deleted.count) && (
+            <>
+              :
+              {!!stats.added.count && (
+                <span className={classes.headerFilesAdded}>
+                  {' +'}
+                  {stats.added.count} ({readableBytes(stats.added.size)})
+                </span>
+              )}
+              {!!stats.deleted.count && (
+                <span className={classes.headerFilesDeleted}>
+                  {' -'}
+                  {stats.deleted.count} ({readableBytes(stats.deleted.size)})
+                </span>
+              )}
+              {warn && <M.Icon style={{ marginLeft: 4 }}>error_outline</M.Icon>}
+            </>
```

Comment:
I would make it a component `<FilesStats addedCount={stats.added.count} deleted={states.deleted.count} />`

Reply:
doesnt seem like that much logic to make it standalone